scanpy/tools/_rank_genes_groups.py

```diff
@@ -35,22 +35,32 @@ def _ranks(X, mask=None, mask_rest=None):
     n_genes = X.shape[1]
 
     if issparse(X):
-        merge = lambda tpl: vstack(tpl).toarray()
-        adapt = lambda X: X.toarray()
+
+        def merge(tpl):
+            return vstack(tpl).toarray()
+
+        def adapt(X):
+            return X.toarray()
+
     else:
         merge = np.vstack
-        adapt = lambda X: X
+
+        def adapt(X):
+            return X
 
     masked = mask is not None and mask_rest is not None
 
     if masked:
         n_cells = np.count_nonzero(mask) + np.count_nonzero(mask_rest)
-        get_chunk = lambda X, left, right: merge(
-            (X[mask, left:right], X[mask_rest, left:right])
-        )
+
+        def get_chunk(X, left, right):
+            return merge((X[mask, left:right], X[mask_rest, left:right]))
+
     else:
         n_cells = X.shape[0]
-        get_chunk = lambda X, left, right: adapt(X[:, left:right])
```

Comment:
We should turn this one off.

Reply:
The lambdas are not the issue. This was likely an unused variable.

Reply:
I'm not sure I understand the change then. How does replacing a `lambda` with a `def` solve this?

Reply:
Oh what the hell. Now I see it. This must have been autopep8. This makes no sense. I will readd the lambdas

Reply:
Actually no, I will not remove this.

```
./scanpy/tools/_rank_genes_groups.py:38:9: E731 do not assign a lambda expression, use a def
./scanpy/tools/_rank_genes_groups.py:39:9: E731 do not assign a lambda expression, use a def
./scanpy/tools/_rank_genes_groups.py:42:9: E731 do not assign a lambda expression, use a def
./scanpy/tools/_rank_genes_groups.py:48:9: E731 do not assign a lambda expression, use a def
./scanpy/tools/_rank_genes_groups.py:53:9: E731 do not assign a lambda expression, use a def
./scanpy/tools/_rank_genes_groups.py:171:13: E731 do not assign a lambda expression, use a def
./scanpy/tools/_rank_genes_groups.py:173:13: E731 do not assign a lambda expression, use a def
./scanpy/tools/_rank_genes_groups.py:743:13: E731 do not assign a lambda expression, use a def
```

Reply:
I'm for turning off `E731`.

Reply:
Context:
```

    The first form means that the name of the resulting function object is specifically 'f' instead of the generic '<lambda>'. This is more useful for tracebacks and string representations in general. The use of the assignment statement eliminates the sole benefit a lambda expression can offer over an explicit def statement (i.e. that it can be embedded inside a larger expression)

Assigning lambdas to names basically just duplicates the functionality of def - and in general, it's best to do something a single way to avoid confusion and increase clarity.

The legitimate use case for lambda is where you want to use a function without assigning it, e.g:
```

I like lambdas as well so you can make the final call.